src/sql/connection/connection.py

```diff
@@ -1187,4 +1174,27 @@ def set_sqlalchemy_isolation_level(conn):
         return False
 
 
+def detect_duckdb_summarize_or_select(query):
+    """
+    Checks if the SQL query is a DuckDB SELECT or SUMMARIZE statement.
+
+    Note:
+    Assumes there is only one SQL statement in the query.
+    """
+    statements = sqlparse.parse(query)
+    if statements:
+        assert len(statements) == 1
```

Comment:
this assert statement will break jupysql. if a user tries to run >1 statement it'll fail here

can we support this fix for >1 statements?

Reply:
I do not see a place where this function would be called with >1 statement, the only place it's used is in `_connection_execute` which has this guard against multiple statements: https://github.com/ploomber/jupysql/pull/895/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1L721-R722

Can you clarify if I am missing something?

Reply:
you're right. let's switch the assert for a `NotImpplementedError` then. asserts are non-informative and should only be used in unit tests

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/895/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R1186-R1187)